api/s2n.h

```diff
@@ -2879,6 +2920,19 @@ S2N_API extern int s2n_connection_get_actual_protocol_version(struct s2n_connect
  */
 S2N_API extern int s2n_connection_get_client_hello_version(struct s2n_connection *conn);
 
+/**
+ * Access the protocol version written in the record header that included the client hello.
+ *
```

Comment:
Nit: kind of awkward phrasing-- the client hello isn't included in the record header :)
```suggestion
 * Access the protocol version from the header of the first record that contained the ClientHello message.
 *
```